apps/generator/lib/conditionalGeneration.js

```diff
@@ -0,0 +1,267 @@
+    return conditionalSubjectGeneration(
+      asyncapiDocument,
+      templateConfig,
+      matchedConditionPath
+    );
+  }
+
+  const parameterValue = await getParameterValue(templateParams, parameter);
+
+  if (parameterValue === undefined) {
+    await handleMissingParameterValue(matchedConditionPath, templateConfig);
+    return false;
+  }
+
+  return validateParameterValue(
+    validation,
+    parameterValue,
+    matchedConditionPath,
+    relativeSourceDirectory,
+    relativeTargetFile,
+    targetDir,
+    templateConfig
+  );
+};
+
+/**
+ * Determines whether a file should be conditionally included based on the provided subject expression
+ * and optional validation logic defined in the template configuration.
+ *
+ * @param {Object} asyncapiDocument - The parsed AsyncAPI document instance used for context evaluation.
+ * @param {Object} templateParams - The template parameters passed by the user (e.g. server selection).
+ * @param {Object} templateConfig - The configuration object that contains `conditionalFiles` rules.
+ * @param {String} relativeSourceFile - The relative path to the source file being evaluated.
+ * @param {String} relativeSourceDirectory - The relative path to the directory of the source file.
+ * @returns {Boolean} - Returns `true` if the file should be included; `false` if it should be skipped.
+ */
+async function conditionalFilesGenerationDeprecatedVersion (
+  asyncapiDocument,
+  templateParams,
+  templateConfig,
+  relativeSourceFile
+) {
+  const fileCondition = templateConfig.conditionalFiles?.[relativeSourceFile];
+  if (!fileCondition || !fileCondition.subject) {
+    return true; 
+  }
+
+  const { subject, validate } = fileCondition;
+
+  const server = templateParams.server && asyncapiDocument.servers().get(templateParams.server);
+  const context = {
+    ...asyncapiDocument.json(),
+    server: server ? server.json() : undefined,
+    info: asyncapiDocument.info()?.json(),
+    channels: asyncapiDocument.channels() || undefined,
+    components: asyncapiDocument.components()?.json(),
+  };
+
+  const source = jmespath.search(context, subject);
+  if (!source) {
+    log.debug(logMessage.relativeSourceFileNotGenerated(relativeSourceFile, subject));
+    return false;
+  }
+
+  if (validate && !validate(source)) {
+    log.debug(logMessage.conditionalFilesMatched(relativeSourceFile));
+    return false;
+  }
+
+  return true;
+};
+
+/**
+ * Asynchronously removes the parent directory of the specified target file, 
+ * if it exists and is a directory.
+ *
+ * @param {string} relativeTargetFile - The relative path to the target file, used to locate its parent directory.
+ * @param {string} targetDir - The base directory where the target file resides.
+ * @returns {Promise<void>} A promise that resolves when the parent directory is removed, or if it does not exist.
+ */
+const removeParentDirectory = async (relativeTargetFile, targetDir) => {
```

Comment:
why any removals are necessary in this functionality

in general it should be much easier:
- condition met, generate
- condition not met, skip processing

Reply:
thanks @derberg changed the method name